### PR TITLE
differentiates mech punches between durand and gygax + removes auto punch from leg acurator

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
balance: gygax now has force 25 instead of 30
balance: gygax inflicts knockdown instead of unconscious
balance: leg acurator doesnt auto punch
/:cl:

gygax punch stun lock is aids, this gives the people being punched a chance of fight back

### CODE REVIEW

**Comment by Dawson1917** [comment]
Remove the auto-attack from leg actuators and you will be a hero

**Comment by AcapitalA** [comment]
Any nerf to the gygax leg actuators is a good nerf. The gygax's ability should be focused on rapid movement rather than kool-aid manning into people, and that's coming from me.

**Comment by FloranOtten** [comment]
One small step for mech, one giant leap for Durand kind. Knockdown is still annoying, because you lose items, but no longer can gygaxes just bumrush into nukies and utterly destroy them.

**Comment by bigfatbananacyclops** [comment]
gygax is incredibly weak even if it doesnt allow the chance for people to fight back, its designed to be killed via range

**Comment by zxaber** [comment]
I was honestly toying with the idea of just removing mech punches altogether. I can't imagine any realistic scenario where we'd want to be swinging around expensive and probably delicate experimental weapons as if they were wooden clubs.

More on topic of this PR, I 100% agree that we should remove the auto-attack from leg actuators. The gygax ability should be about rapid threat response and repositioning, not becoming a steel bull.

**Comment by Tlaltecuhtli** [comment]
you mean speed boost without everything else?

**Comment by Dawson1917** [comment]
> 
> 
> you mean speed boost without everything else?

Gygaxes should be quick at responding to events across the station, like zxaber said, and excel at being a fast, mobile gun platform. The durand should be the opposite - a beefy, armored tank for dealing with stationary threats like blobs, xeno nests, or cult bases. So yeah pretty much make the gygax's ability just be a speed boost - it's strong enough doing just that because its leg actuator speed is faster than human running speed.

**Comment by Arathianen** [comment]
This is a good nerf and one that (mostly) brings the mechs to appropriate balance. The gygax needs a shitton of diamonds to be made and should be tougher.

One further change I would like to see is the gygax and the durand switch places in the tech tree with the gygax being higher and durand being the "budget mech".

@zxaber That is going too far. The charged gygax should punch automatically. It's its big strength and one with a hefty downside in a much lower health pool and armour. Gygaxes are also faaar more expensive to make in that they require diamonds which are a much more limited resource than titanium. This nerf is enough.

Honestly, mechs are in a good place with this in general.

**Comment by Dawson1917** [comment]
Doing 5 less damage doesn't stop anyone from using mechs to instant-stun by running into people and then continuing to melee them or back off to riddle them with bullets @Arathianen 
The auto-hit on movement IS why gygaxes are cancer in the first place

**Comment by zxaber** [comment]
I'm also not a huge fan of how quickly gygaxs can just delete walls by walking into them with the autopuncher.

**Comment by Tlaltecuhtli** [comment]
i was thinking of a mech katana like in those animes that does high dismember damage

**Comment by necromanceranne** [comment]
Gygax needs an identity shift and bad. This is just a gutting of what it has now and frankly it doesn't seem like a healthy change because the ability and the mech itself is the proble. I don't mean to say it shouldn't necessarily get changed at all, but it highlights that the gygax has exactly one benefit and that's the leg actuators. It is of the standard crew combat mechs (excluding the HONK) the softest in terms of durability and survivability and only favoured for its speed and autopunches. You could justifiably say it doesn't NEED weapons because of actuators, and sits at odds with the other mechs in that they're generally vulnerable and unable to very easily respond to attacks while emp'd. The gygax does not care. It spears into the ion shots and smacks you in the face.

I should mention that no Tlal, this doesn't mean you can still lie down against a gygax and think you're going to survive. I know that's your trick but knockdown still slaps the gun out of your hand and actuators can still close distance and getting stepped on is entirely possible and still a good enough tactic. Just because your one maneuver isn't working as well as you'd like doesn't mean you need to buff it at the exclusion of a more healthy and balanced change which probably actually involves a overhaul of the mech itself.

**Comment by Tlaltecuhtli** [comment]
gygax speciality is being able to go where the help is needed with leg acurators not stun locking people

**Comment by necromanceranne** [comment]
Honestly the focus on leg actuators is the mistake I feel. I personally feel it would be better to just boost the baseline speed of the mech and make the ability something more unique or interesting. Even if that means some stat adjustments to make it a less durable but faster mech. If it has to be the fast mech it should survive on being smart and moving properly over just tanking shots like an ape, which is entirely the job of the durand. I always liked one suggestion see a lot. A dash of some kind. Something that requires a bit of thought to use. Could be something to explore given you want to make changes.

**Comment by Arathianen** [comment]
I am strongly opposed to remove the auto-punch.

**Comment by Dawson1917** [comment]
> 
> 
> I am strongly opposed to remove the auto-punch.
> 
> It makes the mech trash for what it costs.
> 
> Bad change.

The Lore(tm) designates the gygax as a fast mech for private military forces and corporations. You can adjust the materials costs so the chicken legs gygax doesn't cost more than the actual walking tank that is a durand. I probably will if this much-deserved nerf goes through anyway.

**Comment by Tlaltecuhtli** [comment]
what do you want the ability to do then? ¯\_(ツ)_/¯

**Comment by Arathianen** [comment]
The """lore""" argument is a meme, let's not go there.

Gygax has a clear niche. It is the fast,  lighter, melee mech that costs a shitton.

This nerf makes it absolutely trash for no good reason. The sleep was OP, but removing punching is going WAY too far. If you will nerf the mech to the fucking ground like that, also adjust the cost at the very least.

**Comment by Cyprex** [comment]
Should still flatten walls, autopunching crew is silly though.

**Comment by Tlaltecuhtli** [comment]
hit them manually

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -111,7 +111,7 @@
 	var/thrusters_active = FALSE
 	var/defense_mode = FALSE
 	var/leg_overload_mode = FALSE
-	var/leg_overload_coeff = 100
+	var/leg_overload_coeff = 80
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/mecha/mecha.dm]: what does this do

also lol this is on base mech?

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 2]
```diff
@@ -347,8 +347,10 @@
 			var/dmg = rand(M.force/2, M.force)
 			switch(M.damtype)
 				if("brute")
-					if(M.force > 20)
+					if(M.force > 35) // durand and other heavy mechas
 						Unconscious(20)
+					else if(M.force > 20 && !IsKnockdown()) // lightweight mechas like gygax
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/human/human_defense.dm]: can you avoid mechapunch if you rest lol?

Comment (Tlaltecuhtli) [reply code/modules/mob/living/carbon/human/human_defense.dm]: yes you avoid being knocked down if you crawl (the damge is same)

Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/carbon/human/human_defense.dm]: tactical resting is still alive lmfao

**Comment by Arathianen** [comment]
@ExcessiveUseOfCobblestone this PR should not be merged without adding a balancing to the mech material requirements (preferably by replacing diamonds with something else).

This is a massive nerf to a mech that is exceedingly expensive. It makes it markedly worse than a durand. If it's going to be the budget mech, it should be _the budget mech_ and not the massively expensive, nigh irreplaceable mech that it currently is.

The OP has already said he believes the cost would need a rebalance but he will do it "later if someone else doesn't". Deleting the mech for weeks until someone else bothers to actually balance it out is bad. Because this PR deletes the mech as in any way useful.

**Comment by Dawson1917** [comment]
@Arathianen If Tlalt's nerf gets merged I will literally adjust the mats cost of the gygax **TODAY**. There's no need to blow it out of proportion. The gygax already has a powerful use and that is being a supremely fast gun platform. Just use its leg actuators to do laps around people while shooting them to pieces, rather than the cancer of "run into people and let the mech auto-kill for me". It is by no means useless.

**Comment by Arathianen** [comment]
You are aware ion disables all weapons, right?

**Comment by Dawson1917** [comment]
> 
> 
> You are aware ion disables all weapons, right?

image
An ion bolt only disables mech weapons for 1 second. Each hit. And they don't stack. You have to be under a _hail_ of ion bolts, all hitting, for it to mean anything during a fight. If you find yourself in that situation, congrats, you've learnt that the ion gun is a counter for mechs. But under any other circumstances, your weapon being disabled for exactly 1 second is very minimal. You can also still punch in that time.

**Comment by Arathianen** [comment]
Woops, that's embarrassing. Someone didn't read the changelogs where ions now disable mech weapons for 8 seconds.

**Comment by necromanceranne** [comment]
Pretty sure it should three or five seconds and not one second. You can spread out the ion shots.

The hail of ion shots does something fair more significant. You start taking critical failures.

**Comment by zxaber** [comment]
It's three seconds. Or, at least, it was when I did the PR.

Ions also don't deal critical failures.

**Comment by necromanceranne** [comment]
No but the ammo capacity and damage mixed with battery loss does. They are better in every way for killing mechs just because of how quickly you can apply the damage. They are very good counters and I don't believe Dawson when he says they're useless. They are, however, not bery good against gygaxes because of autopunchies.

**Comment by FloranOtten** [comment]
Removing autopunches is one step too far. Why ever make it? It's crazily expensive, and even without being expensive, it would still mean that there is no mech capable of fighting an ion rifle. Being unable to fight for up to 3 seconds per shot, while the opponent spams a thing that drains the battery, and the only way to catch up is a thing that drains the battery, is bad enough. Now imagine they can just utterly dodge the attacks.

**Comment by zxaber** [comment]
Is it really that game-breaking that mechs get a bad matchup?

**Comment by Arathianen** [comment]
It's fairly game breaking that an advanced piece of tech that costs a ton of roundstart can be countered completely by any chucklefuck that gets to the armoury roundstart or quickly print a midgame ion gun.

People need to stop treating mechs like roboticists spawn with a couple of them already in the mechbay. They are expensive, high tech tier and cost a ton of resources. They are not supposed to die to a light breeze.

**Comment by ExcessiveUseOfCobblestone** [comment]
Sec doesn't _need_ a roundstart ion now they're craftable and borgs, the more common reason to use ions, have other counters. Especially so when EMP can be damning far more than the war on stuns [tm]

**Comment by necromanceranne** [comment]
Ion carbines are quite late tech and are portable and ergonomic. I don't honestly see how that is logical at all. In general aoe emps are stronger against sec by turning off their equipment and fucking up the environment, but focused emps are strong against only mechs and borgs. The ion rifle isn't reasonably portable either. It's an emergency weapon and is pretty solidly balanced as such. I honestly can't see how a big bulky weapon like the rifle is even remotely worth having over the more dependable shotguns while deploying emp grenades for an antagonist, and security doesn't want to use emp grenades because the collateral is too much and fucks them more than their target.

Unless you want the things which literally cannot be stunned or slowed down short of selective tools (mechs and borgs), and presently have very strong stun tools which either can't be overcome (stun arm for borgs and mech punches) or require powergaming to prevent (borg flash/service borg spraycan and mech flashbangs), it seems utterly ridiculous to strip security of their single answer without first making at least borgs not the powerhouses they are.

**Comment by ExcessiveUseOfCobblestone** [comment]
Hard Stuns should be/are transitioning out though...

**Comment by necromanceranne** [comment]
That's literally all that affects borgs. They suffer zero downsides compared to carbons and aren't even vulnerable to damage from a completely unarmed individual. There is no other method of subduing a borg. You'd need to fundamentally change borgs to fit that criteria. And that's a long way off and definitely should come before the ion rifle is removed.

**Comment by ExcessiveUseOfCobblestone** [comment]
> presently have very strong stun tools which either can't be overcome (stun arm for borgs and mech punches) or require powergaming to prevent (borg flash/service borg spraycan and mech flashbangs)

Yes, which is why we should move out of hardstuns FROM borgs.

> There is no other method of subduing a borg except hard stuns and death and lockdown, the latter of which can be removed as an option entirely by wily sabotage.

Yes, which is why as hard stuns TO borgs get removed they'll need to be more susceptible to things we normally associate with carbons such as damage slowdown.

That said, even with roundstart ion removal, there's still several ways to disable a borg. I'd rather have consumable EMP nades/non-renewable "emergency" version (HoS can keep his baby ion) roundstart > bulky ion in techwebbies > upgradable to ioncarbon in techwebbies.

**Comment by ExcessiveUseOfCobblestone** [comment]
what does the `leg...coef` DO?

**Comment by Tlaltecuhtli** [comment]
it makes leg overload consume slighty less when it hasnt tier 4 parts

**Review by Zaffinomlb — COMMENTED** [review]
The deletion of the "bumpsmash" is the most trash part of this PR. The cost has still yet to be lower. Nerfing the knock down timers or the effects rather then the auto smash function would be far more balanced then just nerfing the whole mech into the ground with no survivors. 

The cost was never even balanced and it honestly has made this mech rather useless. It is just a mech to be skipped now. This PR frankly smells of pro durand bias. high dex over str or stam. "Gygax punch stun lock is aids, this gives the people being punched a chance of fight back".

With this PR can any mech break down doors or walls now without manually clicking? Said clicking action is delayed so any effective breaching of doors or walls, much less things like firelocks or blastdoors. will take far to long to be effective anymore.

File: code/game/mecha/mecha_actions.dm [hunk 3]
```diff
@@ -187,13 +187,11 @@
 	chassis.log_message("Toggled leg actuators overload.", LOG_MECHA)
 	if(chassis.leg_overload_mode)
 		chassis.leg_overload_mode = 1
-		chassis.bumpsmash = 1
 		chassis.step_in = min(1, round(chassis.step_in/2))
 		chassis.step_energy_drain = max(chassis.overload_step_energy_drain_min,chassis.step_energy_drain*chassis.leg_overload_coeff)
 		chassis.occupant_message("<span class='danger'>You enable leg actuators overload.</span>")
 	else
 		chassis.leg_overload_mode = 0
-		chassis.bumpsmash = 0
```
Comment (Zaffinomlb) [inline code/game/mecha/mecha_actions.dm]: Honestly you are a bad person for doing this.

**Comment by zxaber** [comment]
You _do_ have the ability to make special breaching missiles now. Sure, they take a slot, but otherwise they get the job done just as well.

**Comment by Zaffinomlb** [comment]
Does that use a weapons slot? Also does that require a special research?

**Comment by Zaffinomlb** [comment]
The most optimal meta load out for gygax as I knew it was relay, armor type, weapon. 

The only thing I have thought of in lieu of these poorly thought out changes would be relay, rockets, gun weapon.

You can still break down normal walls but literally anything else takes too long.

Dark gygax might as well be not worth taken now also. The TC cost of that should also be rebalanced because of this poorly thought out PR.

**Comment by zxaber** [comment]
>poorly thought out PR.
Look, man, I get that you don't like this change. But doubling down on that isn't getting us anywhere. I, for one, am getting sick of people abusing tools that let them delete the walls of entire departments while also being difficult to stop (and I'd love to see hulk mutation get the same treatment for the same reason).

Instead of complaining about a change that is already done, we should look into other ways to buff the Gygax.

**Comment by Zaffinomlb** [comment]
What do you have in mind, firstly the cost should be reduced clearly. Maybe bringing the smash back but adding a cool down. Reducing the stun time of the smash make it more like a shove effect. 

Removing it out right is a mistake and that is one of the buffs that needs to be brought back.

**Comment by zxaber** [comment]
For one, whatever the buff is, I _don't_ think it should allow the Gygax to tear down walls like tissue paper. I also disagree with an unarmed Gygax being able to just mow down people.

In all the world of possible buffs, is there not a single other thing we could give to the Gygax? Does it without a doubt _have_ to be a melee attack?

Anyway, this PR is complete, this conversation should be discussed in the Ideas subforum.

**Comment by Anonymous583** [comment]
i ded

**Pull request merged**